File: solutions/python/two-sum.py
def two_sum(nums, target):
    seen = {}
    for i, val in enumerate(nums):
        diff = target - val
        if diff in seen:
            return [seen[diff], i]
        seen[val] = i
    return []

print(two_sum([2,7,11,15], 9))

File: solutions/cpp/two_sum.cpp
#include <iostream>
#include <unordered_map>
#include <vector>
using namespace std;

vector<int> twoSum(vector<int>& nums, int target) {
    unordered_map<int,int> mp;
    for (int i = 0; i < nums.size(); i++) {
        int diff = target - nums[i];
        if (mp.count(diff)) return {mp[diff], i};
        mp[nums[i]] = i;
    }
    return {};
}

int main() {
    vector<int> v = {2,7,11,15};
    auto r = twoSum(v, 9);
    cout << r[0] << "," << r[1];
}
